### PR TITLE
[server] Removes momentjs and replaces it with dayjs

### PR DESCRIPTION
**Pull request by easyCZ**
## Description

Because we don't want to use it due it being deprecated: https://github.com/gitpod-io/gitpod/pull/13543#issuecomment-1265675920

## Related Issue(s)

Fixes #

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`

### CODE REVIEW

File: components/ee/db-sync/package.json [hunk 1]
```diff
@@ -21,7 +21,6 @@
     "@gitpod/gitpod-db": "0.1.5",
     "@gitpod/gitpod-protocol": "0.1.5",
     "argparse": "^2.0.1",
-    "moment": "^2.21.0",
```
Comment (easyCZ) [inline components/ee/db-sync/package.json:24]: Was specified here but not used.

**Comment by laushinka** [comment]
What's the best way to test that this migration didn't break anything? Just make sure `dashboard` builds?

**Comment by easyCZ** [comment]
Basically. `dayjs` is API compatible with `momentjs`. I can also run a preview env for this just to sanity check from the reviewer side too.

**Comment by laushinka** [comment]
`dashboard` builds and I checked in the preview all the views that was affected by the change.
I can approve the implementation but will hold depending on whether we would like a further discussion on the use of `dayjs` in these discussions: [[1](https://github.com/gitpod-io/gitpod/pull/13568)][[2](https://github.com/gitpod-io/gitpod/pull/13543)].
I see the merit of using the Date object instead of being dependent on another library, but it was also a bit of a pain to use it for our logic. 

/hold

**Review by laushinka — APPROVED** [review]

**Comment by easyCZ** [comment]
This change alone does not make the problem any worse. We still use a library to do this albeit a non-deprecated one. And `dayjs` is smaller than moment so there's already a win in doing this.

If we want to go to native Date, it should be independent of this change, in my opinion.

**Comment by laushinka** [comment]
> This change alone does not make the problem any worse. We still use a library to do this albeit a non-deprecated one. 

Excellent point. Using an external library and using a deprecated one are two different things. If we decide to go with the Date object we'll create a new ticket.

/unhold

**Pull request merged**